velox/type/Filter.h

```diff
@@ -1878,6 +1895,20 @@ class BytesValues final : public Filter {
     }
   }
 
+  std::string lower() {
+    return lower_;
+  }
+
+  bool isSingleValue() const override {
+    return isSingleValue_;
+  }
+
+  bool testBloomFilter(
+      std::unique_ptr<AbstractBloomFilter>& bloomFilter,
+      const velox::Type& type) const override {
+    return bloomFilter->mightContain(lower_);
```

Comment:
I don't think this is correct, you need to loop over all `values_` here

Reply:
We can apply bloom filter only for exact lookup(equality predicate) and not range of values, in which case `lower_` and `upper_` will be the same. `isSingleValue` method is used for determining this. The check for single value is made in the caller method `rowGroupMatches()`

Reply:
If it is only supposed to be called with single value filter, please document it in the interface method and check the filter is single valued inside the method (remove `isSingleValue()`), instead of exposing this implementation details to file readers:

```c++
bool testBloomFilter(
      std::unique_ptr<AbstractBloomFilter>& bloomFilter,
      const velox::Type& type) const override {
  return lower_ < upper_ || bloomFilter->mightContain(lower_);
}
```

Reply:
`isSingleValue()` was already implemented as a public method before my changes for some Filter's sub classes. I just added an interface method in Filter class added overrides since this can be used for this PR's purpose.

Reply:
I will move the call to isSingleValue() inside the testBloomFilter().